submissions/okkkko/js_oop/app.js

```diff
@@ -1,87 +1,102 @@
-    if (Math.abs(this.x - this.player.x)<BUG_SIZE  &&
-        Math.abs(this.y - this.player.y)<BUG_SIZE) {
-        alert('You lose!');
-        this.player.resetPosution();
-        };
+
+const Enemy = function (x, y, speed, player) {
+  Character.call(this, x, y);
+  this.sprite = "images/enemy-bug.png";
+  this.speed = speed;
+  this.player = player;
 };
-Enemy.prototype.render = function() {
-    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+Enemy.prototype = Object.create(Character.prototype);
+Enemy.prototype.update = function (dt) {
+  this.x += this.speed * dt;
+  if (this.x > CANVAS_WIDTH) {
+    this.x = BUGS_START_POSITION.x;
+  }
+  this.handleCollision();
 };
-
-const Player = function(x,y){
-    this.sprite = 'images/char-boy.png';
-    this.x = x;
-    this.y = y;
+Enemy.prototype.handleCollision = function () {
+  if (
+    Math.abs(this.x - this.player.x) < BUG_SIZE &&
+    Math.abs(this.y - this.player.y) < BUG_SIZE
+  ) {
+    alert("You lose!");
+    this.player.resetPosution();
+  }
 };
-Player.prototype.update = function(dt) {
-    if (this.y < 0){
-        setTimeout(()=>{
-            alert("You win!");
-            this.resetPosution();
-        },0)
-    };
+
+const Player = function (x, y) {
+  Character.call(this, x, y);
+  this.sprite = "images/char-boy.png";
 };
-Player.prototype.render = function() {
-    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+Player.prototype = Object.create(Character.prototype);
+Player.prototype.update = function (dt) {
+  if (this.y < 0) {
+    setTimeout(() => {
+      alert("You win!");
+      this.resetPosution();
+    }, 0);
+  }
 };
-Player.prototype.handleInput = function(key){
-    if (key==="left" && this.x>0){
-        this.x-=STEP_X;
-    }if (key === "right" && this.x<CANVAS_WIDTH-STEP_X){
-        this.x+=STEP_X;
-    }if (key === "down" && this.y<PLAYER_START_POSUTION.y){
-        this.y+=STEP_Y;
-    }if (key === "up" && this.y>0){
-        this.y-=STEP_Y;
-    };
+Player.prototype.handleInput = function (key) {
+  if (key === "left" && this.x > 0) {
+    this.x -= STEP_X;
+  }
+  if (key === "right" && this.x < CANVAS_WIDTH - STEP_X) {
+    this.x += STEP_X;
+  }
+  if (key === "down" && this.y < PLAYER_START_POSITION.y) {
+    this.y += STEP_Y;
+  }
+  if (key === "up" && this.y > 0) {
+    this.y -= STEP_Y;
+  }
 };
-Player.prototype.resetPosution = function(){
-    this.y = PLAYER_START_POSUTION.y;
-    this.x = PLAYER_START_POSUTION.x;
+Player.prototype.resetPosution = function () {
+  this.y = PLAYER_START_POSITION.y;
+  this.x = PLAYER_START_POSITION.x;
 };
 
-let player = new Player(PLAYER_START_POSUTION.x,PLAYER_START_POSUTION.y);
-BUGS_START_POSUTION.y.forEach(function(y){
-    allEnemies.push(new Enemy(BUGS_START_POSUTION.x, y, getRandomSpeed(MIN_SPEED,MAX_SPEED),player))
- });
+const player = new Player(PLAYER_START_POSITION.x, PLAYER_START_POSITION.y);
+const allEnemies = BUGS_START_POSITION.y.map(
```

Comment:
Here will be better to do map from rows

Also you can do automatic rows calculation)
example 
```
const enemyRows = numberOfRows => Array.from({ length: numberOfRows }, (_, i) => i + 1); 
// will return array of length you provide and fill it with numbers 1 to numberOfRows 

const allEnemies = enemyRows(10).map(row => {
   return new Enemy( BUGS_START_POSITION.x, BUGS_START_POSITION.y(row) .....)
})

```